### PR TITLE
GSW as a service

### PR DESCRIPTION
**Pull request by ZZwatermelon**
A script which creates a .service file, enabling GSW to be run as a service

### CODE REVIEW

**Comment by AarC10** [comment]
image

?

**Comment by AarC10** [comment]
Nvm. Figured out why. Maybe add to the README for setting this up. Need to test further too btw. I see it has the correct perms now, but when running mem_view, it still complains about failing to open file, and I cant run sims atm

**Comment by ZZwatermelon** [comment]
>  when running mem_view, it still complains about failing to open file
I believe I fixed this

**Comment by AarC10** [comment]
Awesome! That seems better. Only comments are to update the README with instructions for setting it up as a service and  I'll approve it for merging

File: README.md [hunk 1]
```diff
@@ -13,3 +13,17 @@ You can always run the GSW service by doing a `./gsw_service` after building. Fo
...
+gsw_service must be built prior to the script being run (and it must exist for the service to work).
+
+Once the script has been run, start the service with:
+sudo systemctl start gsw
+
+### Running the Service
+Stop the service with:
+sudo systemctl stop gsw
+
+If you want the sevrice to run on startup:
```
Comment (AarC10) [inline README.md:28]: Typo here

File: README.md [hunk 2]
```diff
@@ -13,3 +13,17 @@ You can always run the GSW service by doing a `./gsw_service` after building. Fo
 ### Unit Tests
 There are several unit tests that can be ran. You can do a `go test ./...` from the root project directory to execute all tests. It is also recommended to run with the -cover
 flag to get coverage statements.
+
+## Create Service Script (for linux)
```
Comment (AarC10) [inline README.md:17]: Capitalize Linux

File: README.md [hunk 3]
```diff
@@ -13,3 +13,17 @@ You can always run the GSW service by doing a `./gsw_service` after building. Fo
...
+The script must be run from the /scripts directory.
+gsw_service must be built prior to the script being run (and it must exist for the service to work).
+
+Once the script has been run, start the service with:
+sudo systemctl start gsw
```
Comment (AarC10) [inline README.md:22]:
```suggestion
`sudo systemctl start gsw`
```

File: README.md [hunk 4]
```diff
@@ -13,3 +13,17 @@ You can always run the GSW service by doing a `./gsw_service` after building. Fo
...
+
+### Running the Service
+Stop the service with:
+sudo systemctl stop gsw
```
Comment (AarC10) [inline README.md:26]:
```suggestion
`sudo systemctl stop gsw`
```

File: README.md [hunk 5]
```diff
@@ -13,3 +13,17 @@ You can always run the GSW service by doing a `./gsw_service` after building. Fo
...
+
+If you want the sevrice to run on startup:
+sudo systemctl enable gsw
```
Comment (AarC10) [inline README.md:29]:
```suggestion
`sudo systemctl enable gsw`
```

File: README.md [hunk 6]
```diff
@@ -13,3 +13,17 @@ You can always run the GSW service by doing a `./gsw_service` after building. Fo
...
 flag to get coverage statements.
+
+## Create Service Script (for linux)
+The script must be run from the /scripts directory.
+gsw_service must be built prior to the script being run (and it must exist for the service to work).
+
+Once the script has been run, start the service with:
+sudo systemctl start gsw
+
+### Running the Service
```
Comment (AarC10) [inline README.md:24]: Should also include that you can check the status of gsw by doing
`sudo systemctl status gsw`

**Review by AarC10 — CHANGES_REQUESTED** [review]

**Review by AarC10 — APPROVED** [review]

**Comment by AarC10** [comment]
MURGE

**Pull request merged**